source/merge.js

```diff
@@ -1,10 +1,14 @@
+function clone(obj) {
+    return Object.setPrototypeOf(Object.assign({}, obj), Object.getPrototypeOf(obj));
```

Comment:
`setPrototypeOf` is supposedly a very heavy operation:

![image](https://user-images.githubusercontent.com/3869469/78068114-89146c00-73a0-11ea-8cf2-1f2b6fc896da.png)

Can we do something like:

```javascript
const OBJ_PROTO = Object.getPrototypeOf({});

function clone(obj) {
	const objProto = Object.getPrototypeOf(obj);
    return objProto === OBJ_PROTO
		? Object.assign({}, obj)
		: Object.setPrototypeOf(Object.assign({}, obj), objProto);
}
```